api/db/migrations/20200317112058_add_unique_constraint_on_userid_certificationcourseid_table_assessments.js

```diff
@@ -0,0 +1,62 @@
+const TABLE_NAME = 'assessments';
+const COLUMN_USER_ID = 'userId';
+const COLUMN_CERTIFICATION_COURSE_ID = 'certificationCourseId';
+
+const ASSESSMENTS_BY_CERTIFICATION_COURSES_CTE =
+  `ASSESSMENTS_BY_CERTIFICATION_COURSES AS (
+      SELECT "as"."id" as assessment_id,
+        COUNT("as"."id") OVER (PARTITION BY "as"."certificationCourseId") AS cnt_assessments_per_certification_course,
+        ROW_NUMBER() OVER (PARTITION BY "as"."certificationCourseId" ORDER BY "as"."state", "as"."createdAt" DESC) as priority
+      FROM "assessments" as "as"
+      WHERE "as"."certificationCourseId" IS NOT NULL
+    )`;
+
+const ASSESSMENTS_TO_DELETE_ID_CTE =
+  `ASSESSMENTS_TO_DELETE_ID AS (
+      SELECT assessment_id as asid
+      FROM ASSESSMENTS_BY_CERTIFICATION_COURSES
+      WHERE cnt_assessments_per_certification_course > 1
+      AND priority <> 1
+    )`;
+
+const ASSESSMENT_RESULTS_TO_DELETE_ID_CTE =
+  `ASSESSMENT_RESULTS_TO_DELETE_ID AS (
+      SELECT "asr"."id" as asrid
+      FROM "assessment-results" as "asr"
+      WHERE "asr"."assessmentId" IN (SELECT asid FROM ASSESSMENTS_TO_DELETE_ID)
+    )`;
```

Comment:
Je suggère plutôt d'utiliser une table temporaire pour ça, ça évitera du travail redondant à PG…

Reply:
Ok je teste. Je t'apporterai  le comparatif de temps d'exécution sur la réplication pour la curiosité. Mais oui, c'est plus élégant la solution de table temporaire.
Sachant que ce SQL + l'alter table prend 50 secondes sur réplication.

Reply:
…et pas vraiment besoin d'une table temporaire ni de CTE si ça se réduit à un `UPDATE` sur l'unique table `assessments`…

Reply:
Got it.

Reply:
Après tout ceci me pose la question de quand est-ce qu'on supprimera les données détachées (le temps de prescription quoi). C'est dommage de se traîner des données mortes au bout d'un certain temps écoulé j'imagine.